File: src/main/java/br/com/leonardoferreira/jirareport/controller/BoardStatusController.java
package br.com.leonardoferreira.jirareport.controller;

import br.com.leonardoferreira.jirareport.service.BoardService;
import java.util.Set;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/boards/{boardId}/statuses")
public class BoardStatusController {

    @Autowired
    private BoardService boardService;

    @GetMapping
    public Set<String> findByBoardId(@PathVariable final Long boardId) {
        return boardService.findStatusFromBoardInJira(boardId);
    }

}

File: application/src/main/kotlin/br/com/jiratorio/controller/BoardStatusController.kt
package br.com.jiratorio.controller

import br.com.jiratorio.service.ExternalBoardService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/boards/{boardId}/statuses")
internal class BoardStatusController(
    private val externalBoardService: ExternalBoardService
) {

    @GetMapping
    fun findByBoardId(@PathVariable boardId: Long): Set<String> =
        externalBoardService.findAllPossibleColumns(boardId)

}
